controllers/events.js

```diff
@@ -0,0 +1,171 @@
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The saved room data in JSON format.
+ * @throws {Error} If an error occurs while creating the room document.
+ */
+const createRoom = async (req, res) => {
+  const { name, description, region } = req.body;
+  const payload = { name, description, region };
+  try {
+    const roomData = await apiService.post("/rooms", payload);
+    const savedRoomData = await eventQuery.createRoom(roomData);
+    return res.status(200).json(savedRoomData);
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't create room. Please try again later",
+    });
+  }
+};
+
+/**
+ * Retrieves all rooms that match the query parameters provided in the HTTP request query string.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The rooms data in JSON format.
+ * @throws {Error} If an error occurs while retrieving the rooms data.
+ */
+const getAllRooms = async (req, res) => {
+  const { enabled, hits, offset } = req.query;
+  try {
+    const start = offset || "";
+    const roomsData = await apiService.get(`/rooms?limit=${hits}&enabled=${enabled}&start=${start}`);
+    return res.status(200).json(roomsData);
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't get rooms. Please try again later",
+    });
+  }
+};
+
+/**
+ * Generates a token for the specified room and user information.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} An object containing the generated token and a success message in JSON format.
+ * @throws {Error} If an error occurs while generating the token.
+ */
+const joinRoom = async (req, res) => {
+  const { roomId, userId, role } = req.body;
+  const payload = { room_id: roomId, user_id: userId, role };
+  try {
+    const token = tokenService.getAuthToken(payload);
+    res.status(200).json({
+      token: token,
+      msg: "Token generated successfully!",
+      success: true,
+    });
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+};
+
+/**
+ * Retrieves room details by ID and returns JSON response
+ *
+ * @async
+ * @function
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ * @returns {Object} - JSON response containing room details or error message
+ * @throws {Object} - The JSON response with an error message if an error occurred if room retrieval fails.
+ */
+const getRoomById = async (req, res) => {
+  const roomId = req.params.id;
+  const enabled = req.body.enabled;
```

Comment:
Yes, enabled is used to get the active or inactive room because the APIs are different is 100ms. 
I didn't understand what confusion is it creating. Please explain.